src/main/java/net/fabricmc/loader/launch/knot/MixinServiceKnot.java

```diff
@@ -173,6 +181,32 @@ public IContainerHandle getPrimaryContainer() {
 
 	@Override
 	public InputStream getResourceAsStream(String name) {
+		if (name.startsWith(FabricMixinBootstrap.MOD_PREFIX)) {
+			int secondColon = name.indexOf(":", FabricMixinBootstrap.MOD_PREFIX.length());
+
+			if (secondColon > 0) {
+				String modid = name.substring(FabricMixinBootstrap.MOD_PREFIX.length(), secondColon);
+				String path = name.substring(secondColon + 1);
+
+				Optional<ModContainer> mc = FabricLoader.getInstance().getModContainer(modid);
+				if (!mc.isPresent()) {
+					return null;
+				}
+				ModContainer mod = mc.get();
+				Path pathInMod = mod.getPath(path);
+				try {
+					InputStream stream = Files.newInputStream(pathInMod);
+					// FIXME: Unlike ClassLoader.getResourceAsStream() this
+					// won't auto-close the returned input stream.
+					// Which is a problem as MixinConfig doesn't close this itself.
+					return stream;
```

Comment:
I *think* this might actually be fine as `ClassLoader.getResourceAsStream()` only closes input streams after it has been closed - and as we keep the classloaders around forever (and only drop them when the application itself terminates) this might be ok?